clients/client-s3/test/unit/dotSegmentInUriLabel.spec.ts

```diff
@@ -0,0 +1,72 @@
+/// <reference types="mocha" />
+import { HttpHandler, HttpRequest, HttpResponse } from "@smithy/protocol-http";
+import { HttpHandlerOptions } from "@smithy/types";
+import { S3 } from "../../src/S3";
+import chai from "chai";
+import chaiAsPromised from "chai-as-promised";
+
+chai.use(chaiAsPromised);
+const { expect } = chai;
+
+/**
+ * Throws an expected exception that contains the serialized request.
+ */
+class EXPECTED_REQUEST_SERIALIZATION_ERROR extends Error {
+  constructor(readonly request: HttpRequest) {
+    super();
+  }
+}
+
+class RequestSerializationTestHandler implements HttpHandler {
+  async handle(request: HttpRequest, options?: HttpHandlerOptions): Promise<{ response: HttpResponse }> {
+    throw new EXPECTED_REQUEST_SERIALIZATION_ERROR(request);
+  }
+  updateHttpClientConfig(key: never, value: never): void {}
+  httpHandlerConfigs() {
+    return {};
+  }
+}
+
+describe("Dot Segment in URI Label", () => {
+  it("S3PreservesLeadingDotSegmentInUriLabel", async () => {
+    const client = new S3({
+      requestHandler: new RequestSerializationTestHandler(),
```

Comment:
is this test only running against the browser dist? 

There should be explicit mock credentials to avoid the default provider chain executing, resulting in a lot of failed requests/file-reads.

Reply:
No. It's running as a unit test on Node.js

https://github.com/aws/aws-sdk-js-v3/blob/e116b4c2388d60586acf6210bc1c3a10b932d6df/clients/client-s3/package.json#L19

Reply:
could you set

```
credentials: {
  accessKeyId: "mockAccessKeyId",
  secretAccessKey: "mockSecretAccessKey",
};
```

to avoid default credential chain execution?

Reply:
Done in [`feb6a80` (#6262)](https://github.com/aws/aws-sdk-js-v3/pull/6262/commits/feb6a80235f56c426790844fce218e99f605757d)